h2o-core/src/main/java/hex/ModelMetricsBinomial.java

```diff
@@ -197,6 +206,11 @@ public static class MetricBuilderBinomial<T extends MetricBuilderBinomial<T>> ex
         // Compute log loss
         _logloss += w * MathUtils.logloss(err);
       }
+
+      if(m != null && m.getClass().toString().contains("Generic")) {
```

Comment:
This looks weird to me. I would prefer not to hardwire some behavior based on the class name.

Would it make sense to add a new predicate e.g. `supportsLikelihood()` so you communicate the intent of the condition. I am unsure why Generic models should be different than others. Or I guess you could use something like `m instanceof GenericModel` but I'd prefer if you consider some alternative approach that would be more in line with OOP.